R/summarystatsteststatistics.R

```diff
@@ -0,0 +1,127 @@
+SummaryStatsTestStatistics <- function(jaspResults, dataset, options, ...) {
+  
+  # Create main table
+  if (is.null(jaspResults[["testStatTable"]])) {
+    jaspResults[["testStatTable"]] <- .testStatMainTable(options)
+  }
+  
+  return()
+}
+
+.testStatMainTable <- function(options) {
+  
+  table <- createJaspTable(title = gettext("Test Statistics"))
+  table$dependOn(c("testType", "testStatistic", "df", "df1", "df2", "alternative"))
+  
+  # Add columns based on test type
+  table$addColumnInfo(name = "testType", title = gettext("Test"), type = "string")
+  table$addColumnInfo(name = "statistic", title = gettext("Statistic"), type = "number")
+  
+  if (options$testType %in% c("t", "chisq")) {
+    table$addColumnInfo(name = "df", title = gettext("df"), type = "integer")
+  }
+  
+  if (options$testType == "f") {
+    table$addColumnInfo(name = "df1", title = gettext("df1"), type = "integer")
+    table$addColumnInfo(name = "df2", title = gettext("df2"), type = "integer")
+  }
+  
+  table$addColumnInfo(name = "pvalue", title = gettext("p"), type = "pvalue")
+  
+  # Calculate p-value
+  results <- .calculatePValue(options)
+  
+  if (!is.null(results$error)) {
+    table$setError(results$error)
+    return(table)
+  }
+  
+  # Create row data
+  rowData <- list(
+    testType = .getTestTypeLabel(options$testType),
+    statistic = options$testStatistic
+  )
+  
+  if (options$testType %in% c("t", "chisq")) {
+    rowData$df <- options$df
+  }
+  
+  if (options$testType == "f") {
+    rowData$df1 <- options$df1
+    rowData$df2 <- options$df2
+  }
+  
+  rowData$pvalue <- results$pvalue
+  
+  table$addRows(rowData)
+  
+  return(table)
+}
+
+.calculatePValue <- function(options) {
+  
+  # Input validation
+  if (is.null(options$testStatistic) || !is.finite(options$testStatistic)) {
+    return(list(error = gettext("Test statistic must be a valid number")))
+  }
+  
+  testStat <- options$testStatistic
+  
+  tryCatch({
+    if (options$testType == "z") {
+      # Z-test
+      if (options$alternative == "twoSided") {
+        pvalue <- 2 * (1 - stats::pnorm(abs(testStat)))
+      } else if (options$alternative == "greater") {
+        pvalue <- 1 - stats::pnorm(testStat)
+      } else { # less
+        pvalue <- stats::pnorm(testStat)
+      }
+    } else if (options$testType == "t") {
+      # t-test
+      if (options$df < 1) {
+        return(list(error = gettext("Degrees of freedom must be positive")))
```

Comment:
this error check are unneccessary because the input value is restricted via the qml menu (there is a min limit of 1)